### PR TITLE
Added CaveWall entity

### LINKED ISSUE #7: Cave Walls
Cave Walls
Properties:
- Follows group properties similar to Floaty Space Blocks
- Isn't solid and becomes invisible when entered, similar to Fake Walls
- When exited from (leaving their hitbox) they become visible again, similar to Exit Blocks without becoming solid
- Enough depth that it is above Fake Walls so that both can be used

What it would do is provide a cover to everything inside of the "cave" unless you are inside it, from things like watchtowers or camera triggers
Will be used for (hopefully) puzzle tower theme

_Requested in collabcord_

### PR DESCRIPTION
**Pull request by Brokemia**
The entity appears solid when the player is not in it, but fades out once a player enters it, similarly to how FakeWalls work. It then fades back in after the player leaves it again.
Adjacent CaveWalls blend together
Closes #7

### CODE REVIEW

File: CaveWall.cs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+                Rectangle rectangle = new Rectangle(GroupBoundsMin.X / 8, GroupBoundsMin.Y / 8, (GroupBoundsMax.X - GroupBoundsMin.X) / 8 + 1, (GroupBoundsMax.Y - GroupBoundsMin.Y) / 8 + 1);
+                Level level = SceneAs<Level>();
+                Rectangle tileBounds = level.Session.MapData.TileBounds;
+                VirtualMap<char> virtualMap = level.SolidsData.Clone();
+                foreach (CaveWall item in Group) {
+                    int num = (int) (item.X / 8f) - level.Session.MapData.TileBounds.X;
+                    int num2 = (int) (item.Y / 8f - level.Session.MapData.TileBounds.Y);
+                    int num3 = (int) (item.Width / 8f);
+                    int num4 = (int) (item.Height / 8f);
```
Comment (Cruor) [inline CaveWall.cs]: Make these variable names less decompilery, like in the foreach right below.

File: CaveWall.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
+﻿using Celeste.Mod.Entities;
+using Microsoft.Xna.Framework;
+using Monocle;
+using System;
+using System.Collections.Generic;
+
+namespace Celeste.Mod.CollabUtils2 {
```
Comment (Cruor) [inline CaveWall.cs]: Wrong namespace should be `Celeste.Mod.SpringCollab2020.Entities`.
Move CaveWall.cs into the `Entities` folder as well.

**Comment by Cruor** [comment]
Also missing tooltips for the entity.

**Comment by Brokemia** [comment]
All of those comments have been fixed

**Comment by 0x0ade** [comment]
I was just gonna comment on the unnecessary (and slightly inconsistent) whitespace between the field definitions, and the fact that the property definition could be one-lined. Those are very minor annoyances though and you've pushed a new commit while I was looking at your code 😅 

Gonna verify that the issues are indeed fixed and merge. Good job! 👍

**Comment by 0x0ade** [comment]
To clarify, I'm not going to wait for a rereview from Cruor as he's streaming right now, and his comments have been addressed.

**Pull request merged**